### PR TITLE
Feature/repo 5395 backport new query acc optimisations

### PR DESCRIPTION
**Pull request by n-insaidoo**
This PR represents the result of the code backport from [feature/repo-5354_demo-a](https://github.com/Alfresco/alfresco-enterprise-repo/tree/feature/repo-5354_demo-a). The new changes include: better stats reporting via API and a point of extension to support the denormalised owner column feature which is mostly implemented on the ent. repo version ([see companion PR](https://github.com/Alfresco/alfresco-enterprise-repo/pull/78)).

### CODE REVIEW

**Review by cmshields — CHANGES_REQUESTED** [review]
Nana, please see the comment about the comments.

Could you look at the failing test in AppContextExtraTestSuite and get the build green before I approve?

File: repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/SingleTaskRestartableWatch.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package org.alfresco.repo.search.impl.querymodel.impl.db;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
```
[the remark this answers is not in the document]
Comment (n-insaidoo) [reply repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/SingleTaskRestartableWatch.java:6]: Little oversight there, sorted it out now.

[the remark this answers is not in the document]
Comment (n-insaidoo) [reply repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/SingleTaskRestartableWatch.java]: @cmshields the failing tests are fixed now

**Review by cmshields — APPROVED** [review]
LGTM

**Pull request merged**